src/gpodder/download.py

```diff
@@ -277,7 +277,13 @@ def retrieve_resume(self, url, filename, reporthook=None, data=None):
             try:
                 resp.raise_for_status()
             except HTTPError as e:
-                raise gPodderDownloadHTTPError(url, resp.status_code, str(e))
+                if auth is not None:
+                    # Try again without authentification (bug 1296)
+                    self.channel.auth_username = None
+                    self.channel.auth_password = None
```

Comment:
This disables authentication in the channel and causes a failure if channel is updated without restarting gpodder.

Reply:
Oops, it causes permanent failure even after restarting gpodder.

Reply:
Thanks for the speedy review, as I said it was a rather naive fix but did not take into account how the download function is used in other parts of the program (as you discovered).
I guess the problem lies in overwriting the `self.channel.auth_username`, I assumed it was temporary but seems like I overwrite the storage. Guess what is required is a more temporary way of telling it to run without `auth` set.
Adding an additional optional argument to `retrieve_resume` named something like try_again_without_authentication would prevent this from happening, do you think  that kind of change would be a working/acceptable solution?

Reply:
I haven't looked at retrieve_resume yet, but if it directly uses the auth fields in channel then a parameter would be the best way. Perhaps name it disable_auth or disable_authentication, because the callsite is performing the "try again" part.

Reply:
Changed the method of retrying, fixed the typo.

Reply:
LGTM. I'm going to test it for a few days and then merge if no issues pop up.